debian/Dockerfile

```diff
@@ -1,5 +1,5 @@
 FROM dwolla/jenkins-agent-nucleus AS nucleus
-FROM openjdk:8-jre
+FROM openjdk:8-jre-slim-buster
```

Comment:
Can we test this before merging to make sure there's nothing weird in Buster that will cause problems for us?

Reply:
Probably a good idea, especially since there's only `slim` options available, not sure what we're losing fully with that.  I'll try a few chef/nvm things locally to see if they still work as expected.

Reply:
Tested with a chef and nvm repo, added a fix for https://github.com/Dwolla/jenkins-agent-docker-nvm to build correctly since we're changing the shell.

Reply:
Can you elaborate on what you changed for the shell stuff, and why it helps?

Reply:
Sure, we were hitting https://github.com/debuerreotype/docker-debian-artifacts/issues/46, so I added the fix they recommend.  Looks to be a common issue with the slim docker images of debian.